Source/ARTRest+Private.h

```diff
@@ -78,4 +78,13 @@ NS_ASSUME_NONNULL_BEGIN
 
 @end
 
+@interface ARTRestInternal (Storage)
+
++ (dispatch_queue_t)storageQueue;
+
+- (nullable NSString *)apnsDeviceToken;
+- (void)setAndPersistAPNSDeviceToken:(nullable NSString *)deviceToken;
+
+@end
```

Comment:
I'm not 100% sure what the purpose of this new, synchronised storage is. But, just going by symmetry, would it not need to also include `setAndPersistIdentityTokenDetails:`?

Reply:
> what the purpose of this new, synchronised storage is
It was discussed in slack here https://ably-real-time.slack.com/archives/CSQEKCE81/p1638465303293500?thread_ts=1638354865.280100&cid=CSQEKCE81

> would it not need to also include setAndPersistIdentityTokenDetails:
Agreed, but it wasn't used anywhere, so I decided to implement only the bare minimum.

Reply:
Sorry, perhaps I'm not understanding - what do you mean by "it wasn't used anywhere"? `setAndPersistIdentityTokenDetails` is still called in `ARTPushActivationState.m`.

Reply:
> It was discussed in slack here

OK, thanks for the link. I think I understand the motivation for the new queue a bit better now, but is it directly relevant to this PR, which is about resetting device details? It seems to me like it might be a separate change

Reply:
> It seems to me like it might be a separate change.

Please see this comment https://github.com/ably/ably-cocoa/pull/1253#discussion_r773931486

Reply:
> `setAndPersistIdentityTokenDetails` is still called in `ARTPushActivationState.m`

Yes, I should inspect more places where storage queue access should be implemented, thanks.

Reply:
I've spent a while trying to understand this storage queue better, but it might be best if we have a call to talk about it, to prevent this becoming a long back-and-forth. What do you think?

But, here's where my current thinking is at. I still don't understand why it's necessary to introduce a new queue in order to achieve the overall objective of this pull request.

Looking at what's contained in the `ARTRestInternal (Storage)` category:

- `apnsDeviceToken` / `setAndPersistAPNSDeviceToken` – access to the APNS device token was not synchronised in the past, why does the fact that we might now reset the device introduce a need to synchronise it?
- `resetDevice` – this method is already being called on a serial queue because it's only called inside a push activation state machine transition method, so why do we need an extra serial queue?

It might be that, if you agree with my [analysis](https://github.com/ably/ably-cocoa/pull/1253#discussion_r773872600) that we don't need to put the device into an intermediate "reset" state, then some of this might simplify when you make that change, too.

Reply:
> then some of this might simplify

Sure, upon splitting this PR into two, this one is going to contain only the very first [commit](https://github.com/ably/ably-cocoa/pull/1253/commits/a240961239e9bab77eddf31cb8cd6402fe8378fe) or so.

> we don't need to put the device into an intermediate "reset" state

I agreed with the splitting this PR, we still need something that marks the device as "reset", no? Lets discuss it in a call.